### PR TITLE
Fixes CA1854 replacing unrelated dictionaries with matching signatures

### PR DESCRIPTION
**Pull request by Miizukii**
Changed Analyzer to first find ContainsKey() and than find all usages with the same unchanged key and dictionary reference in the guarded path.

Fixes: #6256, #6390, #6280

### CODE REVIEW

File: src/NetAnalyzers/VisualBasic/Microsoft.NetCore.Analyzers/Runtime/BasicPreferDictionaryTryGetValueFixer.vb [hunk 1]
```diff
@@ -16,60 +16,123 @@ Namespace Microsoft.NetCore.VisualBasic.Analyzers.Runtime
...
+            Dim replaceFunction =
+                    Async Function(ct As CancellationToken) As Task(Of Document)
+                        Dim editor = Await DocumentEditor.CreateAsync(document, ct).ConfigureAwait(False)
+                        Dim generator = editor.Generator
+
+                        Dim tryGetValueAccess = generator.MemberAccessExpression(containsKeyAccess.Expression,
+                                                                                 TryGetValue)
+                        Dim keyArgument = containsKeyInvocation.ArgumentList.Arguments.FirstOrDefault()
+                        Dim valueAssignment = generator.LocalDeclarationStatement(dictionaryValueType, Value).
+                                WithLeadingTrivia(SyntaxFactory.EndOfLine(Environment.NewLine)).
```
Comment (Youssef1313) [inline src/NetAnalyzers/VisualBasic/Microsoft.NetCore.Analyzers/Runtime/BasicPreferDictionaryTryGetValueFixer.vb:87]: Try to use `ElasticCarriageReturnLineFeed`.
I don't think `Environment.NewLine` should be used.

Comment (Miizukii) [reply src/NetAnalyzers/VisualBasic/Microsoft.NetCore.Analyzers/Runtime/BasicPreferDictionaryTryGetValueFixer.vb]: For some reason this adds a line break but also an empty newline after `Dim value As Integer`.
Is there a better way to do it or is that how visual basic wants to be formatted?

File: src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/Runtime/CSharpPreferDictionaryTryGetValueFixer.cs [hunk 2]
```diff
@@ -22,24 +23,57 @@ public sealed class CSharpPreferDictionaryTryGetValueFixer : PreferDictionaryTry
...
                 return;
-            }
 
             Document document = context.Document;
             SyntaxNode root = await document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
 
-            var dictionaryAccess = root.FindNode(dictionaryAccessLocation.SourceSpan, getInnermostNodeForTie: true);
-            if (dictionaryAccess is not ElementAccessExpressionSyntax
-                || root.FindNode(context.Span) is not InvocationExpressionSyntax { Expression: MemberAccessExpressionSyntax containsKeyAccess } containsKeyInvocation)
+            if (root.FindNode(context.Span) is not InvocationExpressionSyntax
```
Comment (mavasani) [inline src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/Runtime/CSharpPreferDictionaryTryGetValueFixer.cs:32]: Why was `getInnermostNodeForTie: true` removed?

**Comment by Miizukii** [comment]
is there still something that needs to be changed?
the failing check is unrelated `coverlet: Unable to read beyond the end of the stream.` bug.

**Comment by Miizukii** [comment]
@Youssef1313 or @mavasani could one of you do another review. Thanks

**Comment by mavasani** [comment]
@buyaa-n Can you please review? Thanks!

**Comment by mavasani** [comment]
@Miizukii for resolving merge conflicts

**Review by buyaa-n — APPROVED** [review]
Thank you @Miizukii! LGTM

**Pull request merged**